pennylane/gradients/jvp.py

```diff
@@ -304,3 +322,12 @@ def _single_measurement_zero(m, tangent):
     res = qml.math.convert_like(np.zeros(dim), tangent)
     res = qml.math.cast_like(res, tangent)
     return res
+
+
+def _shots_copies(shot_vector):
+    """Helper function to determine the number of copies from a shot vector."""
+    if isinstance(shot_vector[0], tuple):
```

Comment:
I think this code would break for `shot_vector = [(2,10), 1000, 100]`. The first element is a `tuple`, but not a `ShotTuple`.

How about
```py
from pennylane._device import ShotTuple
if any(isinstance(shot_comp, ShotTuple) for shot_comp in shot_vector):
```

Reply:
Done thanks 👍